### PR TITLE
initial commit to use newtabprefprovider

### PR DESCRIPTION
**Pull request by sarracini**

### CODE REVIEW

File: src/js/page.js [hunk 1]
```diff
@@ -145,9 +147,9 @@
       let nodeSelector = "#newtab-grid, #newtab-search-container";
       for (let node of document.querySelectorAll(nodeSelector)) {
         if (aValue) {
```
Comment (marcoscaceres) [inline src/js/page.js]: I wonder if we should rename "aValue", just to give more context of what it is in these kinds of situations.

File: src/js/page.js [hunk 2]
```diff
@@ -145,9 +147,9 @@
       let nodeSelector = "#newtab-grid, #newtab-search-container";
       for (let node of document.querySelectorAll(nodeSelector)) {
```
Comment (marcoscaceres) [inline src/js/page.js]: Maybe shorter: 

``` JS
document.querySelectorAll("#newtab-grid, #newtab-search-container")
  .forEach(node => node.dataset.pageDisabled = !aValue);
```

**Comment by marcoscaceres** [comment]
Some early feedback, looking good. What's the purpose of this thingy?

**Comment by sarracini** [comment]
@marcoscaceres it has two purposes:
1. split out all the pref work needed on the newtab page (enabled/enhanced & pinning) and delegate to a module. that module and that module only will listen for pref changes and emit necessary messages (which will be useful for syncing)
2. as a result this actually fixes the customize panel bustage that was mentioned in issue #104 
:)

feel free to give some feedback - i'll have a proper patch up by thursday for both remote-newtab and for newtab-dev for review

**Comment by marcoscaceres** [comment]
@sarracini thanks for the explanation!

File: src/js/page.js (hunk 2 above)
Comment (sarracini) [inline src/js/page.js]: @marcoscaceres  can't use forEach on querySelectorAll...not an array! :(
could do something like 

``` javascript
var nodeList = document.querySelectorAll("#newtab-grid, #newtab-search-container"); 
var nodeArray = Array.prototype.slice.call(nodeList);
nodeArray.forEach(node => node.dataset.pageDisabled = !aValue);
```

thoughts?

Comment (marcoscaceres) [inline src/js/page.js]: Sent from my iPhone

> On Nov 6, 2015, at 3:28 AM, Ursula Sarracini notifications@github.com wrote:
> 
> In src/js/page.js:
> 
> > @@ -145,9 +147,9 @@
> >        let nodeSelector = "#newtab-grid, #newtab-search-container";
> >        for (let node of document.querySelectorAll(nodeSelector)) {
> > @marcoscaceres can't use forEach on querySelectorAll...not an array! :(
> > could do something like
> 
> var nodeList = document.querySelectorAll("#newtab-grid, #newtab-search-container"); 
> var nodeArray = Array.prototype.slice.call(nodeList);
> nodeArray.forEach(node => node.dataset.pageDisabled = !aValue);
> thoughts?
> 
> Ah yeah, you are right. 
> With Es6, we have 2 better options:
> 
> ``` js
> Array.from(nodeList).forEarch();
> // or
> (...nodeList).forEach()
> ```

So we never need to use the ugly Array.prototype.slice.call(💩) hack thing anymore 🤗.

**Comment by sarracini** [comment]
@marcoscaceres @emtwo @oyiptong r? 
I'm not crazy about how we're setting gNewTab.enabled and gNewTab.enhanced but it works for now

**Comment by marcoscaceres** [comment]
R+ from me. It will do for now till we do the web idl stuff.

**Pull request merged**